actionpack/lib/action_controller/metal/request_forgery_protection.rb

```diff
@@ -240,6 +284,30 @@ def handle_unverified_request
       end
     end
 
+    class SessionStore
+      def fetch(request)
+        request.session[:_csrf_token]
+      end
+
+      def store(request, csrf_token)
+        request.session[:_csrf_token] = csrf_token
+      end
+    end
+
+    class CookieStore
+      def initialize(cookie = :csrf_token)
+        @cookie = cookie
+      end
+
+      def fetch(request)
+        request.cookie_jar[@cookie]
+      end
+
+      def store(request, csrf_token)
+        request.cookie_jar[@cookie] = { value: csrf_token, httponly: true, same_site: :lax }
```

Comment:
Is this potentially overriding a more-strict `same_site` value?

Reply:
Ah yep, that makes sense to me. (I bet _someone_'s going to have a specific security policy that all their cookies have to be strict or something, but given the issue you note, I think we should perhaps navigate that when it arises and we can get some more context.)

Reply:
> I bet someone's going to have a specific security policy that all their cookies have to be strict or something

Yes, there's even a note about that in the Django discussion: https://groups.google.com/g/django-developers/c/3vG7H3kRBZ0

> Django's CSRF implementation usually sets off all kinds of false
alarms in most pen-tester's tools, since it doesn't work exactly the
same way other implementations do, and isn't tied to the session
cookie.

But I think it's actually much better this way.